src/carma_clock.cpp

```diff
@@ -52,6 +52,10 @@ timeStampSeconds CarmaClock::nowInSeconds() const {
 }
 
 void CarmaClock::update(timeStampMilliseconds current_time) {
+    // check for sim time and throw exception if not in sim
+    if (!_is_simulation_mode) {
+        throw std::invalid_argument("Clock is not in simulation mode!");
```

Comment:
Only note would be, do we have some documentation somewhere that indicates that the update method should only be used in simulation mode. Maybe we can add a @throws invalid_argument exception when not in simulation mode to the method comment

Reply:
Nvm I see this is already kind of included in the header method comment. This is fine although using javadoc annotations like @param or @throws helps us generate better documentation with doxygen. This is fine for now.